File: Leetcode-Problems/Max Consecutive Ones.cpp
class Solution
{
public:
    int findMaxConsecutiveOnes(vector<int> &nums)
    {
        int ans = 0;
        int temp = nums[0] == 0 ? 0 : 1;
        for (int i = 1; i < nums.size(); i++)
        {
            if (nums[i] == 1)
            {
                temp++;
            }
            else
            {
                ans = max(ans, temp);
                temp = 0;
            }
        }
        ans = max(ans, temp);
        return ans;
    }
};

File: Leetcode-Problems/Max Consecutive Ones.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int ans = 0, temp = 0;
        for (int i : nums) {
            if (i == 1)
                temp++;
            else {
                ans = Math.max(ans, temp);
                temp = 0;
            }
        }
        ans = Math.max(ans, temp);
        return ans;
    }
}